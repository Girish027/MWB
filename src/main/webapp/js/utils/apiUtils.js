
// Make all ids strings

export const normalizeIds = (item) => {
  const newItem = Object.assign({}, item);
  if (newItem.hasOwnProperty('id')) {
    newItem.id = `${newItem.id}`;
  }
  if (newItem.hasOwnProperty('projectId')) {
    newItem.projectId = `${newItem.projectId}`;
  }
  if (newItem.hasOwnProperty('cid')) {
    newItem.cid = `${newItem.cid}`;
  }
  if (newItem.hasOwnProperty('clientId')) {
    newItem.clientId = `${newItem.clientId}`;
  }
  if (newItem.hasOwnProperty('configId')) {
    newItem.configId = `${newItem.configId}`;
  }

  return newItem;
};

export const normalizeModel = (model) => {
  const newModel = normalizeIds(model);
  if (newModel.datasetIds.length > 1) {
    const newDatasets = [];
    newModel.datasetIds.forEach((element) => {
      newDatasets.push(element.trim());
    });
    newModel.datasetIds = newDatasets;
  }
  return newModel;
};

export const normalizeIdArray = (idArray) => {
  const newIds = [];
  idArray.forEach((id) => {
    newIds.push(`${id}`);
  });

  return newIds;
};
