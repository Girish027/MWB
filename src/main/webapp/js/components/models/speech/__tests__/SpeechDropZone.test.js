import React from 'react';
import { shallow, mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import * as apiUrls from 'utils/apiUrls';
import { SpeechDropzone } from 'components/models/speech/SpeechDropzone';

describe('<SpeechDropzone />', () => {
  let wrapper;

  const { featureFlags } = global.uiConfig;

  const props = {
    showDropbox: true,
    userFeatureConfiguration: featureFlags.DEFAULT,
    dispatch: jest.fn(),
    onChange: jest.fn(),
    onRadioChange: jest.fn(),
    onChangeDigitalHostedUrl: jest.fn(),
  };

  const file = {
    lastModified: '1575018060587',
    lastModifiedDate: 'Fri Nov 29 2019 14:31:00 GMT+0530 (India Standard Time) {}',
    name: 'word_class.txt',
    path: 'word_class.txt',
    size: 3876,
    type: 'text/plain',
    webkitRelativePath: '',
  };

  afterAll(() => {
    jest.clearAllMocks();
  });

  describe('Creating an instance', () => {
    test('should exist', () => {
      wrapper = shallow(<SpeechDropzone
        {...props}
      />);
      expect(wrapper.exists()).toBe(true);
    });
  });

  describe('Snapshots', () => {
    test('renders correctly for default props', () => {
      wrapper = mount(<SpeechDropzone
        {...props}
      />);
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
  });

  describe('Functionality:', () => {
    beforeEach(() => {
      props.onChange = jest.fn();
      props.onRadioChange = jest.fn();
      apiUrls.getDataUrl = jest.fn();
      props.onChangeDigitalHostedUrl = jest.fn();
    });

    describe('onChangeRadioButton', () => {
      test('should call onChange props when bundled data is selected in radio', () => {
        wrapper = shallow(<SpeechDropzone
          {...props}
        />);

        wrapper.instance().onChangeRadioButton('bundled-data');
        expect(props.onRadioChange).toHaveBeenCalledWith({ isUnbundled: false });
      });

      test('should call onChange props when unbundled data is selected in radio', () => {
        wrapper = shallow(<SpeechDropzone
          {...props}
        />);

        wrapper.instance().onChangeRadioButton('unbundled-data');
        expect(props.onRadioChange).toHaveBeenCalledWith({ isUnbundled: true });
      });
    });

    describe('onClickDownload', () => {
      test('should call onChange props when bundled data is selected in radio', () => {
        wrapper = shallow(<SpeechDropzone
          {...props}
        />);

        wrapper.instance().onClickDownload();
        expect(apiUrls.getDataUrl).toHaveBeenCalledWith(apiUrls.pathKey.wordClassDefault);
      });
    });

    describe('saveFile', () => {
      test('should save the text file', () => {
        wrapper = shallow(<SpeechDropzone
          {...props}
        />);

        wrapper.instance().handleFileChoosen = jest.fn();
        wrapper.instance().saveFile([file]);

        let blob = new Blob([file], { type: 'text/plain;charset=utf-8' });
        expect(wrapper.instance().handleFileChoosen).toHaveBeenCalledWith(blob);
      });
    });

    describe('handleFileChoosen', () => {
      test('should read the text file', () => {
        wrapper = shallow(<SpeechDropzone
          {...props}
        />);

        let blob = new Blob([file], { type: 'text/plain;charset=utf-8' });
        wrapper.instance().handleFileChoosen(blob);

        expect(wrapper.instance().fileReader).toEqual(expect.anything());
      });
    });

    describe('onChangeDigitalUrl', () => {
      test('should call onChangeDigitalUrl', () => {
        const evt = {
          target: {
            value: 'http://digitalUrl.com/digital',
          },
        };
        const digitalUrl = evt.target.value;
        wrapper = shallow(<SpeechDropzone
          {...props}
        />);

        wrapper.instance().onChangeDigitalUrl(evt);
        expect(props.onChangeDigitalHostedUrl).toHaveBeenCalledWith(digitalUrl);
      });
    });

    describe('rejectFile', () => {
      test('should call rejectFile', () => {
        wrapper = shallow(<SpeechDropzone
          {...props}
        />);
        wrapper.setState({
          isValid: true,
        });

        wrapper.instance().rejectFile();
        expect(props.onChange).toHaveBeenCalledWith({}, false);
        expect(wrapper.state().isValid).toBe(false);
      });
    });

    describe('handleFileRead', () => {
      test('should properly validate the file when content uploaded is in correct format', () => {
        wrapper = shallow(<SpeechDropzone
          {...props}
        />);

        wrapper.instance().wordClassFile = file;
        wrapper.instance().fileReader = {
          result: '_class_family\n'
          + 'son',
        };
        wrapper.update();
        wrapper.instance().handleFileRead({});

        expect(props.onChange).toHaveBeenCalledWith(file, true);
        expect(wrapper.state().isValid).toBe(true);
      });

      test('should properly validate the file when content uploaded is not in correct format', () => {
        wrapper = shallow(<SpeechDropzone
          {...props}
        />);

        wrapper.instance().wordClassFile = file;
        wrapper.instance().fileReader = {
          result: '_class_family test'
          + 'son',
        };
        wrapper.update();
        wrapper.instance().handleFileRead({});

        expect(props.onChange).toHaveBeenCalledWith({}, false);
        expect(wrapper.state().isValid).toBe(false);
      });

      test('should properly validate the file when content uploaded is not in correct format 2', () => {
        wrapper = shallow(<SpeechDropzone
          {...props}
        />);

        wrapper.instance().wordClassFile = file;
        wrapper.instance().fileReader = {
          result: '_class_family test'
          + 'son'
          + '_class_ newtest',
        };
        wrapper.update();
        wrapper.instance().handleFileRead({});

        expect(props.onChange).toHaveBeenCalledWith({}, false);
        expect(wrapper.state().isValid).toBe(false);
      });
    });
  });
});
