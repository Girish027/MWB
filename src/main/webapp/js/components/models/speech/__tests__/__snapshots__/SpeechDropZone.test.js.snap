// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SpeechDropzone /> Snapshots renders correctly for default props 1`] = `
<SpeechDropzone
  dispatch={[MockFunction]}
  onChange={[MockFunction]}
  onChangeDigitalHostedUrl={
    [MockFunction] {
      "calls": Array [
        Array [
          "undefined/undefined/digital",
        ],
        Array [
          "undefined/undefined/digital",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  onRadioChange={[MockFunction]}
  showDropbox={true}
  userFeatureConfiguration={
    Object {
      "appHelp": "show",
      "createNewVersionModelReviewTab": "show",
      "createNewVersionTrainingConfigTab": "show",
      "datasetDelete": "show",
      "datasetExport": "show",
      "datasetExportById": "show",
      "datasetTemplate": "show",
      "downloadResults": "show",
      "granularMessage": "show",
      "intentGuideContextualActionBar": "show",
      "intentGuideDelete": "show",
      "intentGuideExport": "show",
      "intentGuideTemplate": "show",
      "intentGuideUpdate": "show",
      "kibanaLogs": "show",
      "modelConfigDownload": "show",
      "modelConfigTemplate": "show",
      "modelDelete": "show",
      "modelDownload": "show",
      "modelTrainingOutputs": "show",
      "models": "show",
      "projectDelete": "show",
      "resolveInconsistencies": "show",
      "speechBundledUnbundled": "show",
      "supportLink": "show",
      "tagSearchHelp": "show",
      "transformationHelp": "show",
    }
  }
>
  <div
    id="speech-dropbox"
    style={
      Object {
        "marginTop": "30px",
        "paddingLeft": "30px",
        "paddingRight": "0px",
      }
    }
  >
    <div
      style={
        Object {
          "paddingBottom": "20px",
        }
      }
    >
      <span
        style={
          Object {
            "color": "#6E6E6E",
            "fontSize": "14.5px",
            "fontWeight": "normal",
            "paddingBottom": "5px",
          }
        }
      >
        Upload word class file
      </span>
      <div
        style={
          Object {
            "display": "inline-flex",
            "float": "right",
            "icon": Object {
              "marginTop": "-3px",
              "paddingRight": "5px",
            },
            "label": Object {
              "color": "#6E6E6E",
              "cursor": "pointer",
              "fontSize": "12.5px",
              "fontWeight": "normal",
              "textDecoration": "underline",
            },
          }
        }
      >
        <span
          style={
            Object {
              "marginTop": "-3px",
              "paddingRight": "5px",
            }
          }
        >
          <withTheme(Download)
            fill="#004c97"
            height="13px"
            width="10px"
          >
            <Download
              fill="#004c97"
              height="13px"
              tfsStyle={
                Object {
                  "icons": Object {
                    "common": Object {
                      "borderColor": "#B2B2B2",
                      "color": "#004C97",
                      "colorLight": "#336FAB",
                      "disabledColor": "#E0E0E0",
                      "fillColor": "#FFFFFF",
                    },
                  },
                }
              }
              width="10px"
            >
              <svg
                height="13px"
                viewBox="0 0 12 16"
                width="10px"
              >
                <path
                  d="M5.06 11.483a.83.83 0 0 0 1.165-.003l3.8-3.785c.88-.884-.441-2.21-1.322-1.326L7.058 8.021c-.275.275-.497.182-.497-.205V.938A.936.936 0 0 0 5.626 0a.936.936 0 0 0-.934.938v6.878c0 .389-.221.481-.497.205L2.55 6.369c-.881-.884-2.203.442-1.322 1.326l3.831 3.788zm5.255 2.642H.938a.938.938 0 0 0 0 1.875h9.377a.938.938 0 0 0 0-1.875z"
                  fill="#004c97"
                  fillRule="nonzero"
                />
              </svg>
            </Download>
          </withTheme(Download)>
        </span>
        <div
          onClick={[Function]}
          style={
            Object {
              "color": "#6E6E6E",
              "cursor": "pointer",
              "fontSize": "12.5px",
              "fontWeight": "normal",
              "textDecoration": "underline",
            }
          }
        >
           Download existing 
        </div>
      </div>
    </div>
    <Dropzone
      accept="text/plain"
      acceptStyle={
        Object {
          "borderColor": "#00e676",
        }
      }
      acceptedIcon={
        Object {
          "$$typeof": Symbol(react.memo),
          "compare": null,
          "type": [Function],
        }
      }
      activeStyle={
        Object {
          "borderColor": "#2196f3",
        }
      }
      baseStyle={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fafafa",
          "borderColor": "#eeeeee",
          "borderRadius": 2,
          "borderStyle": "dashed",
          "borderWidth": 2,
          "color": "#bdbdbd",
          "display": "flex",
          "flex": 1,
          "flexDirection": "column",
          "height": "170px",
          "outline": "none",
          "padding": "20px",
          "transition": "border .24s ease-in-out",
        }
      }
      disabled={false}
      icon={[Function]}
      multiple={false}
      rejectFile={[Function]}
      rejectStyle={
        Object {
          "borderColor": "#ff1744",
        }
      }
      removeFile={false}
      saveFile={[Function]}
      shouldUpload={false}
      uploadHandler={[Function]}
    >
      <div
        onBlur={[Function]}
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fafafa",
            "borderColor": "#eeeeee",
            "borderRadius": 2,
            "borderStyle": "dashed",
            "borderWidth": 2,
            "color": "#bdbdbd",
            "display": "flex",
            "flex": 1,
            "flexDirection": "column",
            "height": "170px",
            "outline": "none",
            "padding": "20px",
            "transition": "border .24s ease-in-out",
          }
        }
        tabIndex={0}
      >
        <input
          accept="text/plain"
          autoComplete="off"
          disabled={false}
          multiple={false}
          onChange={[Function]}
          onClick={[Function]}
          style={
            Object {
              "display": "none",
            }
          }
          tabIndex={-1}
          type="file"
        />
        <div
          className="dropzone-content"
          style={
            Object {
              "margin": "auto",
            }
          }
        >
          <div
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
            <div
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              <TextIcon
                fill="#9B9B9B"
                height={48}
                width={48}
              >
                <svg
                  height={48}
                  width={48}
                >
                  <g
                    fill="#9B9B9B"
                    fillRule="nonzero"
                  >
                    <path
                      d="M47.987 22.731a.986.986 0 0 0-.053-.263c-.011-.032-.019-.063-.033-.094a.955.955 0 0 0-.193-.284h-.001L42 16.395v-5.57c0-.023-.011-.042-.013-.064a1.023 1.023 0 0 0-.051-.256c-.011-.032-.019-.063-.034-.094a.994.994 0 0 0-.196-.292l-9-8.976a.97.97 0 0 0-.294-.195c-.03-.014-.06-.022-.091-.032a.94.94 0 0 0-.263-.052C32.039.862 32.021.852 32 .852H7c-.552 0-1 .447-1 .997v14.547L.293 22.088l-.002.002a.977.977 0 0 0-.192.284c-.014.031-.022.062-.033.094a.948.948 0 0 0-.053.263c-.002.022-.013.04-.013.062v18.95c0 .55.448.997 1 .997h5v4.987c0 .55.448.997 1 .997h34c.552 0 1-.447 1-.997V42.74h5c.552 0 1-.447 1-.997v-18.95c0-.022-.011-.04-.013-.062zm-3.401-.935H42v-2.58l2.586 2.58zm-6-11.968H33V4.257l5.586 5.571zM8 2.847h23v7.978c0 .55.448.998 1 .998h8V21.796H8V2.847zm-2 16.37v2.579H3.414L6 19.216zm34 27.512H8v-3.99h32v3.99zm6-5.984H2V23.791h44v16.954z"
                    />
                    <path
                      d="M16.11 38v-8.575h3.045v-1.547h-7.987v1.547h3.059V38h1.883zm5.068 0a.401.401 0 0 0 .28-.102c.075-.067.138-.14.189-.22l2.247-3.584c.023-.042.047-.088.07-.137.023-.049.044-.101.063-.157l2.345 3.878a.925.925 0 0 0 .192.224.455.455 0 0 0 .305.098h1.883l-3.353-5.285 3.262-4.837h-1.806c-.163 0-.296.086-.399.259l-2.107 3.325a1.958 1.958 0 0 0-.175.357l-2.303-3.717a.473.473 0 0 0-.157-.172.575.575 0 0 0-.284-.052h-1.876l3.227 4.914L19.414 38h1.764zm12.768 0v-8.575h3.045v-1.547h-7.987v1.547h3.059V38h1.883z"
                    />
                  </g>
                </svg>
              </TextIcon>
            </div>
            <div
              style={
                Object {
                  "paddingTop": "10px",
                  "textAlign": "center",
                  "whiteSpace": "pre-wrap",
                }
              }
            >
              <div>
                Drag and drop your file
                <div>
                   or 
                </div>
                <div
                  style={
                    Object {
                      "color": "#004c97",
                      "cursor": "pointer",
                    }
                  }
                >
                   BROWSE
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </Dropzone>
    <div>
      <div
        style={
          Object {
            "label": Object {
              "color": "#6E6E6E",
              "fontSize": "14.5px",
              "fontWeight": "normal",
              "marginBottom": "10px",
              "paddingBottom": "5px",
            },
            "marginTop": "20px",
          }
        }
      >
        <div
          style={
            Object {
              "color": "#6E6E6E",
              "fontSize": "14.5px",
              "fontWeight": "normal",
              "marginBottom": "10px",
              "paddingBottom": "5px",
            }
          }
        >
           Output 
        </div>
        <div>
          <withTheme(RadioGroup)
            displayType="stacked"
            labels={
              Array [
                "Bundle GRXML and SLM output",
                "Separate GRXML from SLM output",
              ]
            }
            onChange={[Function]}
            value="unbundled-data"
            values={
              Array [
                "bundled-data",
                "unbundled-data",
              ]
            }
          >
            <RadioGroup
              displayType="stacked"
              images={Array []}
              isDisabled={false}
              labels={
                Array [
                  "Bundle GRXML and SLM output",
                  "Separate GRXML from SLM output",
                ]
              }
              onChange={[Function]}
              showImage={false}
              styleOverride={Object {}}
              tfsStyle={
                Object {
                  "radio": Object {
                    "container": Object {
                      "active": Object {
                        "alignItems": "center",
                        "cursor": "pointer",
                        "display": "flex",
                        "lineHeight": "20px",
                        "margin": "2px",
                      },
                      "disabled": Object {
                        "alignItems": "center",
                        "cursor": "not-allowed",
                        "display": "flex",
                        "lineHeight": "20px",
                        "margin": "2px",
                      },
                      "vertical": Object {
                        "alignItems": "inherit",
                        "flexDirection": "column",
                        "marginLeft": "10px",
                      },
                    },
                    "icon": Object {
                      "activeChecked": Object {
                        "borderColor": "#004C97",
                      },
                      "activeUnchecked": Object {
                        "borderColor": "#B2B2B2",
                      },
                      "disabled": Object {
                        "borderColor": "#E0E0E0",
                      },
                    },
                    "image": Object {
                      "boxShadow": "0px 0px 10px rgba(0, 0, 0, 0.35)",
                      "display": "block",
                      "margin": "20px auto",
                      "maxHeight": "150px",
                      "maxWidth": "200px",
                      "textAlign": "center",
                    },
                    "imageContainer": Object {
                      "checked": Object {
                        "backgroundColor": "#d7d7f2",
                        "border": "1px solid #004C97",
                      },
                      "hover": Object {
                        "border": "1px solid #003467",
                      },
                      "unchecked": Object {
                        "border": "1px solid #E6E6E6",
                      },
                    },
                    "imageContainerChecked": Object {
                      "backgroundColor": "#d7d7f2",
                      "border": "1px solid #004C97",
                    },
                    "input": Object {
                      "active": Object {
                        "cursor": "pointer",
                        "opacity": 0,
                        "position": "absolute",
                        "visibility": "hidden",
                      },
                      "disabled": Object {
                        "cursor": "not-allowed",
                        "opacity": 0,
                        "position": "absolute",
                        "visibility": "hidden",
                      },
                    },
                    "label": Object {
                      "color": "#727272",
                      "fontSize": "14px",
                      "fontWeight": "normal",
                      "lineHeight": "20px",
                      "padding": "0px 0px 0px 10px",
                    },
                  },
                }
              }
              value="unbundled-data"
              values={
                Array [
                  "bundled-data",
                  "unbundled-data",
                ]
              }
            >
              <div
                style={
                  Object {
                    "display": "flex",
                    "flexDirection": "column",
                  }
                }
              >
                <withTheme(Radio)
                  checked={false}
                  disabled={false}
                  key="bundled-data"
                  label="Bundle GRXML and SLM output"
                  onChange={[Function]}
                  showImage={false}
                  styleOverride={
                    Array [
                      Object {},
                    ]
                  }
                  value="bundled-data"
                >
                  <Radio
                    checked={false}
                    disabled={false}
                    label="Bundle GRXML and SLM output"
                    onChange={[Function]}
                    showImage={false}
                    styleOverride={
                      Array [
                        Object {},
                      ]
                    }
                    tfsStyle={
                      Object {
                        "radio": Object {
                          "container": Object {
                            "active": Object {
                              "alignItems": "center",
                              "cursor": "pointer",
                              "display": "flex",
                              "lineHeight": "20px",
                              "margin": "2px",
                            },
                            "disabled": Object {
                              "alignItems": "center",
                              "cursor": "not-allowed",
                              "display": "flex",
                              "lineHeight": "20px",
                              "margin": "2px",
                            },
                            "vertical": Object {
                              "alignItems": "inherit",
                              "flexDirection": "column",
                              "marginLeft": "10px",
                            },
                          },
                          "icon": Object {
                            "activeChecked": Object {
                              "borderColor": "#004C97",
                            },
                            "activeUnchecked": Object {
                              "borderColor": "#B2B2B2",
                            },
                            "disabled": Object {
                              "borderColor": "#E0E0E0",
                            },
                          },
                          "image": Object {
                            "boxShadow": "0px 0px 10px rgba(0, 0, 0, 0.35)",
                            "display": "block",
                            "margin": "20px auto",
                            "maxHeight": "150px",
                            "maxWidth": "200px",
                            "textAlign": "center",
                          },
                          "imageContainer": Object {
                            "checked": Object {
                              "backgroundColor": "#d7d7f2",
                              "border": "1px solid #004C97",
                            },
                            "hover": Object {
                              "border": "1px solid #003467",
                            },
                            "unchecked": Object {
                              "border": "1px solid #E6E6E6",
                            },
                          },
                          "imageContainerChecked": Object {
                            "backgroundColor": "#d7d7f2",
                            "border": "1px solid #004C97",
                          },
                          "input": Object {
                            "active": Object {
                              "cursor": "pointer",
                              "opacity": 0,
                              "position": "absolute",
                              "visibility": "hidden",
                            },
                            "disabled": Object {
                              "cursor": "not-allowed",
                              "opacity": 0,
                              "position": "absolute",
                              "visibility": "hidden",
                            },
                          },
                          "label": Object {
                            "color": "#727272",
                            "fontSize": "14px",
                            "fontWeight": "normal",
                            "lineHeight": "20px",
                            "padding": "0px 0px 0px 10px",
                          },
                        },
                      }
                    }
                    value="bundled-data"
                  >
                    <label
                      data-radium={true}
                      onBlur={[Function]}
                      onFocus={[Function]}
                      onMouseOut={[Function]}
                      onMouseOver={[Function]}
                      style={
                        Object {
                          "alignItems": "center",
                          "cursor": "pointer",
                          "display": "flex",
                          "lineHeight": "20px",
                          "margin": "2px",
                        }
                      }
                    >
                      <input
                        data-radium={true}
                        defaultChecked={false}
                        defaultValue="bundled-data"
                        key=".0"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        style={
                          Object {
                            "cursor": "pointer",
                            "opacity": 0,
                            "position": "absolute",
                            "visibility": "hidden",
                          }
                        }
                        type="radio"
                      />
                      <RadioIcon
                        checked={false}
                        disabled={false}
                        focus={false}
                        hover={false}
                        key=".1"
                        style={
                          Object {
                            "activeChecked": Object {
                              "borderColor": "#004C97",
                            },
                            "activeUnchecked": Object {
                              "borderColor": "#B2B2B2",
                            },
                            "disabled": Object {
                              "borderColor": "#E0E0E0",
                            },
                          }
                        }
                      >
                        <svg
                          height="20px"
                          viewBox="0 0 20 20"
                          width="20px"
                        >
                          <g
                            fill="none"
                            fillRule="evenodd"
                            stroke="none"
                            strokeWidth="1"
                          >
                            <g
                              transform="translate(-200.000000, -824.000000)"
                            >
                              <g
                                transform="translate(200.000000, 824.000000)"
                              >
                                <rect
                                  fill="#FFFFFF"
                                  height="19"
                                  rx="9.5"
                                  stroke="#B2B2B2"
                                  width="19"
                                  x="0.5"
                                  y="0.5"
                                />
                              </g>
                            </g>
                          </g>
                        </svg>
                      </RadioIcon>
                      <span
                        data-radium={true}
                        key=".2"
                        style={
                          Object {
                            "color": "#727272",
                            "fontSize": "14px",
                            "fontWeight": "normal",
                            "lineHeight": "20px",
                            "padding": "0px 0px 0px 10px",
                          }
                        }
                      >
                        Bundle GRXML and SLM output
                      </span>
                    </label>
                  </Radio>
                </withTheme(Radio)>
                <withTheme(Radio)
                  checked={true}
                  disabled={false}
                  key="unbundled-data"
                  label="Separate GRXML from SLM output"
                  onChange={[Function]}
                  showImage={false}
                  styleOverride={
                    Array [
                      Object {},
                    ]
                  }
                  value="unbundled-data"
                >
                  <Radio
                    checked={true}
                    disabled={false}
                    label="Separate GRXML from SLM output"
                    onChange={[Function]}
                    showImage={false}
                    styleOverride={
                      Array [
                        Object {},
                      ]
                    }
                    tfsStyle={
                      Object {
                        "radio": Object {
                          "container": Object {
                            "active": Object {
                              "alignItems": "center",
                              "cursor": "pointer",
                              "display": "flex",
                              "lineHeight": "20px",
                              "margin": "2px",
                            },
                            "disabled": Object {
                              "alignItems": "center",
                              "cursor": "not-allowed",
                              "display": "flex",
                              "lineHeight": "20px",
                              "margin": "2px",
                            },
                            "vertical": Object {
                              "alignItems": "inherit",
                              "flexDirection": "column",
                              "marginLeft": "10px",
                            },
                          },
                          "icon": Object {
                            "activeChecked": Object {
                              "borderColor": "#004C97",
                            },
                            "activeUnchecked": Object {
                              "borderColor": "#B2B2B2",
                            },
                            "disabled": Object {
                              "borderColor": "#E0E0E0",
                            },
                          },
                          "image": Object {
                            "boxShadow": "0px 0px 10px rgba(0, 0, 0, 0.35)",
                            "display": "block",
                            "margin": "20px auto",
                            "maxHeight": "150px",
                            "maxWidth": "200px",
                            "textAlign": "center",
                          },
                          "imageContainer": Object {
                            "checked": Object {
                              "backgroundColor": "#d7d7f2",
                              "border": "1px solid #004C97",
                            },
                            "hover": Object {
                              "border": "1px solid #003467",
                            },
                            "unchecked": Object {
                              "border": "1px solid #E6E6E6",
                            },
                          },
                          "imageContainerChecked": Object {
                            "backgroundColor": "#d7d7f2",
                            "border": "1px solid #004C97",
                          },
                          "input": Object {
                            "active": Object {
                              "cursor": "pointer",
                              "opacity": 0,
                              "position": "absolute",
                              "visibility": "hidden",
                            },
                            "disabled": Object {
                              "cursor": "not-allowed",
                              "opacity": 0,
                              "position": "absolute",
                              "visibility": "hidden",
                            },
                          },
                          "label": Object {
                            "color": "#727272",
                            "fontSize": "14px",
                            "fontWeight": "normal",
                            "lineHeight": "20px",
                            "padding": "0px 0px 0px 10px",
                          },
                        },
                      }
                    }
                    value="unbundled-data"
                  >
                    <label
                      data-radium={true}
                      onBlur={[Function]}
                      onFocus={[Function]}
                      onMouseOut={[Function]}
                      onMouseOver={[Function]}
                      style={
                        Object {
                          "alignItems": "center",
                          "cursor": "pointer",
                          "display": "flex",
                          "lineHeight": "20px",
                          "margin": "2px",
                        }
                      }
                    >
                      <input
                        data-radium={true}
                        defaultChecked={true}
                        defaultValue="unbundled-data"
                        key=".0"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        style={
                          Object {
                            "cursor": "pointer",
                            "opacity": 0,
                            "position": "absolute",
                            "visibility": "hidden",
                          }
                        }
                        type="radio"
                      />
                      <RadioIcon
                        checked={true}
                        disabled={false}
                        focus={false}
                        hover={false}
                        key=".1"
                        style={
                          Object {
                            "activeChecked": Object {
                              "borderColor": "#004C97",
                            },
                            "activeUnchecked": Object {
                              "borderColor": "#B2B2B2",
                            },
                            "disabled": Object {
                              "borderColor": "#E0E0E0",
                            },
                          }
                        }
                      >
                        <svg
                          height="20px"
                          viewBox="0 0 20 20"
                          width="20px"
                        >
                          <g
                            fill="none"
                            fillRule="evenodd"
                            stroke="none"
                            strokeWidth="1"
                          >
                            <g
                              transform="translate(-200.000000, -824.000000)"
                            >
                              <g
                                transform="translate(200.000000, 824.000000)"
                              >
                                <rect
                                  fill="#004C97"
                                  height="19"
                                  rx="9.5"
                                  stroke="#004C97"
                                  width="19"
                                  x="0.5"
                                  y="0.5"
                                />
                                <rect
                                  fill="#FFFFFF"
                                  height="8"
                                  rx="4"
                                  width="8"
                                  x="6"
                                  y="6"
                                />
                              </g>
                            </g>
                          </g>
                        </svg>
                      </RadioIcon>
                      <span
                        data-radium={true}
                        key=".2"
                        style={
                          Object {
                            "color": "#727272",
                            "fontSize": "14px",
                            "fontWeight": "normal",
                            "lineHeight": "20px",
                            "padding": "0px 0px 0px 10px",
                          }
                        }
                      >
                        Separate GRXML from SLM output
                      </span>
                    </label>
                  </Radio>
                </withTheme(Radio)>
              </div>
            </RadioGroup>
          </withTheme(RadioGroup)>
        </div>
      </div>
      <div
        style={
          Object {
            "label": Object {
              "color": "#6E6E6E",
              "fontSize": "14.5px",
              "fontWeight": "normal",
              "marginBottom": "10px",
              "paddingBottom": "5px",
            },
            "marginTop": "20px",
          }
        }
      >
        <div
          style={
            Object {
              "color": "#6E6E6E",
              "fontSize": "14.5px",
              "fontWeight": "normal",
              "marginBottom": "10px",
              "paddingBottom": "5px",
            }
          }
        >
           Digital Model Url 
        </div>
        <withTheme(TextField)
          defaultValue="undefined/undefined/digital"
          name="digital-url"
          onChange={[Function]}
          styleOverride={
            Object {
              "paddingBottom": "15px",
              "width": "100%",
            }
          }
          title="URL of Digital Model for the Speech recognizer to point to"
          type="text"
        >
          <TextField
            alertIconProps={Object {}}
            autoComplete="on"
            checkmarkProps={Object {}}
            defaultValue="undefined/undefined/digital"
            disabled={false}
            invalid={false}
            name="digital-url"
            onChange={[Function]}
            onKeyDown={[Function]}
            showValidCheck={false}
            styleOverride={
              Object {
                "paddingBottom": "15px",
                "width": "100%",
              }
            }
            tfsStyle={
              Object {
                "textField": Object {
                  "container": Object {
                    "cursor": "auto",
                    "display": "inline-block",
                    "width": "250px",
                  },
                  "errorInput": Object {
                    ":active": Object {
                      "border": "1px solid #E03C31",
                    },
                    ":focus": Object {
                      "color": "inherit",
                    },
                    "border": "1px solid #E03C31",
                    "color": "#E03C31",
                  },
                  "helpText": Object {
                    "fontSize": "12px",
                    "height": "13px",
                    "lineHeight": 1,
                    "marginTop": "5px",
                  },
                  "input": Object {
                    ":active": Object {
                      "border": "1px solid #000000",
                    },
                    ":focus": Object {
                      "border": "1px solid #336FAB",
                    },
                    ":hover": Object {
                      "border": "1px solid #000000",
                    },
                    "backgroundColor": "#FFFFFF",
                    "border": "1px solid #B2B2B2",
                    "borderRadius": "4px",
                    "boxSizing": "border-box",
                    "height": "40px",
                    "margin": "0px",
                    "opacity": 1,
                    "outline": "none",
                    "padding": "8px",
                    "width": "100%",
                  },
                },
              }
            }
            title="URL of Digital Model for the Speech recognizer to point to"
            tooltipDirection="bottom"
            tooltipText=""
            tooltipTrigger="click"
            type="text"
          >
            <div
              data-radium={true}
              style={
                Object {
                  "cursor": "auto",
                  "display": "inline-block",
                  "paddingBottom": "15px",
                  "width": "100%",
                }
              }
            >
              <div
                data-radium={true}
                style={
                  Object {
                    "position": "relative",
                  }
                }
              >
                <input
                  autoComplete="on"
                  data-radium={true}
                  defaultValue="undefined/undefined/digital"
                  disabled={false}
                  key=".0"
                  name="digital-url"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#FFFFFF",
                      "border": "1px solid #B2B2B2",
                      "borderRadius": "4px",
                      "boxSizing": "border-box",
                      "height": "40px",
                      "margin": "0px",
                      "opacity": 1,
                      "outline": "none",
                      "padding": "8px",
                      "width": "100%",
                    }
                  }
                  title="URL of Digital Model for the Speech recognizer to point to"
                  type="text"
                />
                <div
                  data-radium={true}
                  key=".1"
                  style={
                    Object {
                      "color": undefined,
                      "fontSize": "12px",
                      "height": "13px",
                      "lineHeight": 1,
                      "marginTop": "5px",
                    }
                  }
                />
              </div>
            </div>
          </TextField>
        </withTheme(TextField)>
        <div />
      </div>
    </div>
  </div>
</SpeechDropzone>
`;
